src/dune_pkg/opam_solver.ml

```diff
@@ -170,11 +185,20 @@ module Context_for_dune = struct
   ;;
 
   let filter_deps t package filtered_formula =
-    let package_is_local =
-      OpamPackage.name package
-      |> Package_name.of_opam_package_name
-      |> Package_name.Map.mem t.local_packages
+    let name = OpamPackage.name package |> Package_name.of_opam_package_name in
+    let filtered_formula =
+      OpamFormula.map_up_formula
+        (fun formula ->
+          match formula with
+          | Atom (pkg, _) ->
+            let name = Package_name.of_opam_package_name pkg in
+            (match Package_name.Map.find t.constraints name with
+             | None -> formula
```

Comment:
Is this how we prevent additional dependencies from being added by the workspace constraints? If so I think this code deserves a comment.